From 69a1e50605bbc4f1b2f61700b1876ff9321c320f Mon Sep 17 00:00:00 2001
From: Ben Allen <bsallen@alcf.anl.gov>
Date: Thu, 8 Mar 2018 13:50:25 -0800
Subject: [PATCH] Update 50-config to detect hostnamectl (eg. systemd) and set
 hostname via /etc/hostname, cleanup rest of script to pass ShellCheck

---
 .../capabilities/provision-vnfs/50-config          | 46 +++++++++++-----------
 1 file changed, 23 insertions(+), 23 deletions(-)

diff --git a/provision/initramfs/capabilities/provision-vnfs/50-config b/provision/initramfs/capabilities/provision-vnfs/50-config
index 9d63c2b..5c75b87 100644
--- a/provision/initramfs/capabilities/provision-vnfs/50-config
+++ b/provision/initramfs/capabilities/provision-vnfs/50-config
@@ -7,37 +7,37 @@
 # required approvals from the U.S. Dept. of Energy).  All rights reserved.
 #
 
-if [ -d "$NEWROOT/etc/sysconfig" ]; then
-    cp /tmp/ifcfg-* $NEWROOT/etc/sysconfig/network-scripts/ 2>/dev/null
+if [ -d "${NEWROOT}/etc/sysconfig/network-scripts" ]; then
+    cp -f /tmp/ifcfg-* "${NEWROOT}/etc/sysconfig/network-scripts/"
+elif [ -d "${NEWROOT}/etc/network" ]; then
+    cp -f /tmp/interfaces "${NEWROOT}/etc/network/"
+fi
 
-    if [ -n "$HOSTNAME" ]; then
-        grep -v "^HOSTNAME=" $NEWROOT/etc/sysconfig/network > /tmp/network_withouthostname
-        cat /tmp/network_withouthostname > $NEWROOT/etc/sysconfig/network
-        rm -f /tmp/network_withouthostname
-        echo "HOSTNAME=$HOSTNAME" >> $NEWROOT/etc/sysconfig/network
-    fi
-else
-    if [ -n "$HOSTNAME" ]; then
-        echo "$HOSTNAME" > $NEWROOT/etc/hostname
+if [ -n "$HOSTNAME" ]; then
+    if [ -f "${NEWROOT}/usr/bin/hostnamectl" ] || [ -f "${NEWROOT}/etc/debian_version" ]; then
+        echo "$HOSTNAME" > "${NEWROOT}/etc/hostname"
+    elif [ -f "${NEWROOT}/etc/sysconfig/network" ]; then
+        sed -i '/^HOSTNAME.*$/d' "${NEWROOT}/etc/sysconfig/network"
+        echo "HOSTNAME=${HOSTNAME}" >> "${NEWROOT}/etc/sysconfig/network"
     fi
-    cp /tmp/interfaces $NEWROOT/etc/network/ 2>/dev/null
 fi
 
-if [ -z "$WWNOFSTABUPDATE" ]; then
-    if [ -f "$NEWROOT/etc/fstab" ]; then
-        > $NEWROOT/etc/fstab_
-        cat $NEWROOT/etc/fstab | while read i; do
-            if [ "x$i" == "x#GENERATED_ENTRIES#" ]; then
-                cat /tmp/fstab >> $NEWROOT/etc/fstab_
+if [ -z "${WWNOFSTABUPDATE}" ]; then
+    if [ -f "${NEWROOT}/etc/fstab" ]; then
+        > "${NEWROOT}/etc/fstab_"
+        while read i; do
+            if [ "${i}" = "#GENERATED_ENTRIES#" ]; then
+                cat /tmp/fstab >> "${NEWROOT}/etc/fstab_"
             else
-                echo "$i" >> $NEWROOT/etc/fstab_
+                echo "${i}" >> "${NEWROOT}/etc/fstab_"
             fi
 
-        done
-        cat $NEWROOT/etc/fstab_ > $NEWROOT/etc/fstab
-        rm -f $NEWROOT/etc/fstab_
+        done < "${NEWROOT}/etc/fstab"
+
+        cp -f "${NEWROOT}/etc/fstab_" "${NEWROOT}/etc/fstab"
+        rm -f "${NEWROOT}/etc/fstab_"
     else
-        cp /tmp/fstab $NEWROOT/etc/fstab
+        cp -f /tmp/fstab "${NEWROOT}/etc/fstab"
     fi
 fi
 
